tensordict/tensorclass.py

```diff
@@ -377,19 +392,31 @@ def wrapper(
         **kwargs,
     ):
 
-        for value, key in zip(args, self.__dataclass_fields__):
-            if key in kwargs:
-                raise ValueError(f"The key {key} is already set in kwargs")
-            kwargs[key] = value
+        if not torch.compiler.is_dynamo_compiling():
+            # zip not supported by dynamo
```

Comment:
It'd be amazing if you create a massive tracker issue in PyTorch. All these are very good onboarding tasks for people that want to learn about dynamo.

Reply:
I'll PM you, I have a write-up I shared internally